src/components/list/list.tsx

```diff
@@ -68,20 +68,16 @@ export class List {
         this.handleAction = this.handleAction.bind(this);
     }
 
-    public componentDidLoad() {
-        this.mdcList = new MDCList(
-            this.element.shadowRoot.querySelector('.mdc-list')
-        );
-
-        this.handleType();
+    public connectedCallback() {
+        this.setup();
     }
 
-    public componentDidUnload() {
```

Comment:
`componentDidUnload` is not really mentioned in the docs any longer, apart from in the lifecycle image. I'm not sure if we should really use it or not, since it has the same behavior as `disconnectedCallback` which is documented. Both `connectedCallback` and `disconnectedCallback` are part of the native web-component API, so I doubt they will be removed from Stencil

I don't think we can have anything as a destructor. We should instead do all setup when the component is connected, and all teardown when it is disconnected. We don't really have to care if the component is destroyed or not, or if it will be reconnected again

Reply:
Ok, that makes sense 👍